benches/mainnet.rs

```diff
@@ -25,35 +25,61 @@ pub fn criterion_benchmark(c: &mut Criterion) {
         // with many dependencies.
         .min(NonZeroUsize::new(8).unwrap());
 
-    common::for_each_block_from_disk(|block, storage| {
+    common::for_each_block_from_disk(|block, in_memory_storage, on_disk_storage| {
         let mut group = c.benchmark_group(format!(
             "Block {}({} txs, {} gas)",
             block.header.number.unwrap(),
             block.transactions.len(),
             block.header.gas_used
         ));
-        group.bench_function("Sequential", |b| {
+        group.bench_function("Sequential (in memory)", |b| {
             b.iter(|| {
                 pevm::execute(
-                    black_box(&storage),
+                    black_box(&in_memory_storage),
                     black_box(&chain),
                     black_box(block.clone()),
                     black_box(concurrency_level),
                     black_box(true),
                 )
             })
         });
-        group.bench_function("Parallel", |b| {
+        group.bench_function("Parallel (in memory)", |b| {
             b.iter(|| {
                 pevm::execute(
-                    black_box(&storage),
+                    black_box(&in_memory_storage),
                     black_box(&chain),
                     black_box(block.clone()),
                     black_box(concurrency_level),
                     black_box(false),
                 )
             })
         });
+        group.bench_function("Sequential (on_disk)", |b| {
+            b.iter(|| {
+                pevm::execute(
+                    black_box(&on_disk_storage),
+                    black_box(&chain),
+                    black_box(block.clone()),
+                    black_box(concurrency_level),
+                    black_box(true),
+                )
+            })
+        });
+        on_disk_storage.clear_cache();
```

Comment:
Intuitively this should be put inside `b.iter` which runs `pevm:execute` several times to measure the average, etc. execution time. Otherwise only the first run will read from disk.